librecad/src/lib/math/lc_linemath.cpp

```diff
@@ -150,15 +152,9 @@ RS_Vector LC_LineMath::calculateEndpointForAngleDirection(double angleValueDegre
  * @return point
  */
 RS_Vector LC_LineMath::getNearestPointOnLine(RS_Line* line, const RS_Vector& coord, bool infiniteLine){
-    if (infiniteLine){
-
-        RS_Vector point1 = line->getStartpoint();
-        RS_Vector point2 = line->getEndpoint();
-        return getNearestPointOnInfiniteLine(coord, point1, point2);
-    }
-    else{
-        return line->getNearestPointOnEntity(coord, true, nullptr);
-    }
+    // For infinite lines, find the nearest point is not limited by start/end points
+    bool onEntity = ! infiniteLine;
+    return line->getNearestPointOnEntity(coord, onEntity, nullptr);
```

Comment:
please take a look to my comment to pen request - I'm not sure that  getNearestPointOnEntity will not return edge endpoints as line is not constructional. Yet buy generic logic of the function, if infiniteLine parameter is true, projection to the ray where the line is located (or projection on construction line) is needed.